src/repository/controllers/user-controller.ts

```diff
@@ -0,0 +1,80 @@
+import argon2 from 'argon2';
+import { Request, Response } from 'express';
+import * as userService from '../services/user-service';
+import { encrypt, decrypt } from '../../crypt';
+
+class UsersController {
+  async createUser(req: Request, res: Response) {
+    req.body.pass = await argon2.hash(req.body.pass);
+    if (!req.body.role) {
+      req.body.role = 'user';
+    }
+    userService
+      .create(req.body)
```

Comment:
That's unnecessary usage of promises. You already did a great start with async/await in the beginning of the function, it will be better to stay with the same approach and keep using `await` further

Reply:
Updated according to comment.
Thanks a lot @rfist !
I appreciate your time, patience and advices